### PR TITLE
Fixes unreachable branch in Simulator#printBehavior

### LINKED ISSUE #979: Unreachable branch after refactoring
class: tlatools/org.lamport.tlatools/src/tlc2/tool/SimulationWorker.java#printBehavior(final TLCState state, final StateVec stateTrace)

In this commit: https://github.com/tlaplus/tlaplus/commit/7973950b96b58194eef0a807be3636eb908792f4 
```
	private final long maxTraceDepth;
```
was changed type to:
```
	private final int maxTraceDepth;
```

But this line was not updated: https://github.com/tlaplus/tlaplus/blob/master/tlatools/org.lamport.tlatools/src/tlc2/tool/Simulator.java#L433

```
		if (this.traceDepth == Long.MAX_VALUE) {
			MP.printMessage(EC.TLC_ERROR_STATE);
			StatePrinter.printStandaloneErrorState(state);
```

### PR DESCRIPTION
**Pull request by FedericoPonzi**
Fixes #979

### CODE REVIEW

**Comment by lemmy** [comment]
Can you come up with a unit test? It's okay if you can't.

File: tlatools/org.lamport.tlatools/test/tlc2/tool/SimulatorTest.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.HashMap;
+
+import static org.junit.Assert.assertTrue;
+
+public class SimulatorTest extends CommonTestCase {
+
+    @Test
+    public void testPrintBehaviorShouldPrintErrorState() throws IOException {
+        MP.setRecorder(recorder);
+        Simulator simulator = new Simulator(new DebugTool(BASE_PATH + "Github726", BASE_PATH + "Github726", new SimpleFilenameToStream(),
```
Comment (FedericoPonzi) [inline tlatools/org.lamport.tlatools/test/tlc2/tool/SimulatorTest.java:23]: it would be nice to include some mocking library...

Comment (lemmy) [reply tlatools/org.lamport.tlatools/test/tlc2/tool/SimulatorTest.java:23]: We just [removed EasyMock](https://github.com/tlaplus/tlaplus/commit/2a942645b7408930841b2271bdfdabf5dd91082d). :-)

**Review by lemmy — APPROVED** [review]
LGTM

**Review by Calvin-L — APPROVED** [review]

**Pull request merged**